test/index.test.js

```diff
@@ -65,3 +65,14 @@ test.serial('postMessage', async t => {
 	t.deepEqual(second.data[2], msg);
 	t.not(second.data[2], msg);
 });
+
+test.serial('close', async t => {
+	const worker = new Worker('./test/fixtures/close.mjs', { type: 'module' });
+	let closed = false;
+	// Not emitted in the browser, just for testing
+	worker.addEventListener('close', e => {
+		closed = true;
+	});
+	await sleep(500);
+	t.is(closed, true, 'should have closed itself');
```

Comment:
It's good to avoid delays for test reliabililty, surely the following can work just as well:

```suggestion
	await new Promise((resolve, reject) => {
	  worker.addEventListener('close', resolve);
	  setTimeout(reject, 500);
	});
```

Reply:
Are we sure that binding the handler inside the promise will be ready in time to catch the close event?

Reply:
Yes, if it was ready in time to set `closed` to `true` then it will be ready in time to resolve the promise. The `new Promise` constructor always runs synchronously and immediately.